lib/portlayer/exec/exec.go

```diff
@@ -15,89 +15,117 @@
-		detail := fmt.Sprintf("expected singular compute resource element, found %d", ccount)
-		log.Errorf(detail)
-		return errors.New(detail)
-	}
+func Init(ctx context.Context, sess *session.Session, source extraconfig.DataSource, _ extraconfig.DataSink) error {
+	var err error
+	initializer.Do(func() {
+		f := find.NewFinder(sess.Vim25(), false)
 
-	cr := VCHConfig.ComputeResources[0]
+		extraconfig.Decode(source, &Config)
 
-	r, err := f.ObjectReference(ctx, cr)
-	if err != nil {
-		detail := fmt.Sprintf("could not get resource pool or virtual app reference from %q: %s", cr.String(), err)
-		log.Errorf(detail)
-		return err
-	}
-	switch o := r.(type) {
-	case *object.VirtualApp:
-		VCHConfig.VirtualApp = o
-		VCHConfig.ResourcePool = o.ResourcePool
-	case *object.ResourcePool:
-		VCHConfig.ResourcePool = o
-	default:
-		detail := fmt.Sprintf("could not get resource pool or virtual app from reference %q: object type is wrong", cr.String())
-		log.Errorf(detail)
-		return errors.New(detail)
-	}
+		log.Debugf("Decoded VCH config for execution: %#v", Config)
+		ccount := len(Config.ComputeResources)
+		if ccount != 1 {
+			err = fmt.Errorf("expected singular compute resource element, found %d", ccount)
+			log.Error(err)
+			return
+		}
 
-	// we want to monitor the resource pool, so create a vSphere Event Collector
-	ec := vsphere.NewCollector(session.Vim25(), VCHConfig.ResourcePool.Reference().String())
+		cr := Config.ComputeResources[0]
+		var r object.Reference
+		r, err = f.ObjectReference(ctx, cr)
+		if err != nil {
+			err = fmt.Errorf("could not get resource pool or virtual app reference from %q: %s", cr.String(), err)
+			log.Error(err)
+			return
+		}
+		switch o := r.(type) {
+		case *object.VirtualApp:
+			Config.VirtualApp = o
+			Config.ResourcePool = o.ResourcePool
+		case *object.ResourcePool:
+			Config.ResourcePool = o
+		default:
+			err = fmt.Errorf("could not get resource pool or virtual app from reference %q: object type is wrong", cr.String())
+			log.Error(err)
+			return
+		}
 
-	// start the collection of vsphere events
-	err = ec.Start()
-	if err != nil {
-		detail := fmt.Sprintf("%s failed to start: %s", ec.Name(), err.Error())
-		log.Error(detail)
-		return err
-	}
+		// we want to monitor the resource pool, so create a vSphere Event Collector
+		ec := vsphere.NewCollector(sess.Vim25(), Config.ResourcePool.Reference().String())
 
-	// create the event manager &  register the existing collector
-	VCHConfig.EventManager = event.NewEventManager(ec)
+		// start the collection of vsphere events
+		err = ec.Start()
+		if err != nil {
+			err = fmt.Errorf("%s failed to start: %s", ec.Name(), err)
+			log.Error(err)
+			return
+		}
 
-	// subscribe the exec layer to the event stream for Vm events
-	VCHConfig.EventManager.Subscribe(events.NewEventType(vsphere.VmEvent{}).Topic(), "exec", eventCallback)
+		// create the event manager &  register the existing collector
+		Config.EventManager = event.NewEventManager(ec)
 
```

Comment:
The subscription to vsphere events has been removed, so OOB activity isn't being reflected in the container cache -- this needs to be added, so events are processed